service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleService.java

```diff
@@ -86,6 +99,25 @@ public Bundle createEncounterBundle(POCDMT000002UK01ClinicalDocument1 clinicalDo
         return bundle;
     }
 
+    private Encounter addEncounterObjects(Encounter encounter, EncounterHelper encounterHelper,
+        POCDMT000002UK01ClinicalDocument1 clinicalDocument, List<QuestionnaireResponse> questionnaireResponseList) {
+        if (encounterHelper.getReferralRequest() != null) {
+            encounter.addIncomingReferral(new Reference(referralRequestMapper.addCondition(
+                encounterHelper.getReferralRequest(), clinicalDocument, encounter, questionnaireResponseList)));
+        }
+
+        if (encounterHelper.getAppointment() != null) {
```

Comment:
If it's `getAppointment()` returns an `Optional` then we should assume it's invalid for it to be null.